app/src/test/java/uk/gov/onelogin/ui/welcome/WelcomeScreenViewModelTest.kt

```diff
@@ -0,0 +1,243 @@
+package uk.gov.onelogin.ui.welcome
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import androidx.lifecycle.Observer
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import uk.gov.android.authentication.AuthenticationError
+import uk.gov.android.authentication.LoginSession
+import uk.gov.android.authentication.TokenResponse
+import uk.gov.android.features.FeatureFlags
+import uk.gov.android.network.online.OnlineChecker
+import uk.gov.onelogin.credentialchecker.BiometricStatus
+import uk.gov.onelogin.credentialchecker.CredentialChecker
+import uk.gov.onelogin.extensions.CoroutinesTestExtension
+import uk.gov.onelogin.extensions.InstantExecutorExtension
+import uk.gov.onelogin.login.LoginRoutes
+import uk.gov.onelogin.login.biooptin.BiometricPreference
+import uk.gov.onelogin.login.biooptin.BiometricPreferenceHandler
+import uk.gov.onelogin.login.ui.welcome.WelcomeScreenViewModel
+import uk.gov.onelogin.login.usecase.VerifyIdToken
+import uk.gov.onelogin.mainnav.nav.MainNavRoutes
+import uk.gov.onelogin.repositiories.TokenRepository
+import uk.gov.onelogin.tokens.usecases.AutoInitialiseSecureStore
+import uk.gov.onelogin.tokens.usecases.GetPersistentId
+
+@OptIn(ExperimentalCoroutinesApi::class)
```

Comment:
~I've seen this a few times, should we not move this to the build.gradle.kts~
I was going to put something about potentially using freeCompilerArgs, but apparently they're going to deprecate it.